automation/src/test/scala/org/broadinstitute/dsde/firecloud/page/workspaces/notebooks/WorkspaceNotebooksPage.scala

```diff
@@ -0,0 +1,207 @@
+
+  override def awaitReady(): Unit = clustersTable.awaitReady()
+
+  override val url: String = s"${Config.FireCloud.baseUrl}#workspaces/$namespace/$name/notebooks"
+
+  private val clustersTable = Table("spark-clusters-table")
+  private val clustersTableTestId = testId("spark-clusters-table")
+  private val clusterErrorMessage = testId("notebooks-error")
+  private val sparkClustersHeader = testId("spark-clusters-title")
+  private val openCreateClusterModalButton: Button = Button("create-modal-button")
+  private val noClustersMessage = "There are no clusters to display."
+
+  //  withNewCluster - open modal, create cluster, wait for it to appear/running state, open delete modal, delete cluster, wait for delete
+  def withNewCluster(clusterName: String,
+                     masterMachineType: String = "n1-standard-4",
+                     masterDiskSize: Int = 500,
+                     workers: Int = 0,
+                     workerMachineType: String = "n1-standard-4",
+                     workerDiskSize: Int = 500,
+                     workerLocalSSDs: Int = 0,
+                     preemptibleWorkers: Int = 0,
+                     extensionURI: Option[GcsPath] = None,
+                     customScriptURI: Option[GcsPath] = None,
+                     labels: Map[String, String] = Map())(testCode: (WorkspaceNotebooksPage) => Any) = {
+
+    openCreateClusterModal.createCluster(clusterName, masterMachineType, masterDiskSize, workers, workerMachineType, workerDiskSize, workerLocalSSDs, preemptibleWorkers, extensionURI, customScriptURI, labels)
+    await text clusterName
+    assert(getClusterStatus(clusterName) == "Creating")
+    logger.info("Creating dataproc cluster " + clusterName)
+    waitUntilClusterIsRunning(clusterName)
+    logger.info("Created dataproc cluster " + clusterName)
+
+    testCode
+
+    openDeleteClusterModal(clusterName).deleteCluster
+    await condition (getClusterStatus(clusterName) == "Deleting")
+    logger.info("Deleting dataproc cluster " + clusterName)
+    waitUntilClusterIsDeleted(clusterName)
+    await text noClustersMessage
+    logger.info("Deleted dataproc cluster " + clusterName)
+  }
+
+  // Goes to a Jupyter page. Currently doesn't return a page object
+  // because Jupyter functionality is tested in leo automated tests.
+  // Just check that it loads
+  def openJupyterPage(clusterName: String): JupyterPage = {
+    val initialWindowHandles = windowHandles
+
+    val clusterLink = Link(clusterName + "-link")
+    await ready clusterLink
+    clusterLink.doClick()
+
+    await condition (windowHandles.size > 1)
+    val jupyterWindowHandle = (windowHandles -- initialWindowHandles).head
+
+    switch to window(jupyterWindowHandle)
+    new JupyterPage().awaitLoaded()
+  }
+
+  private def waitUntilClusterIsRunning(clusterName: String) = {
+    while (getClusterStatus(clusterName) == "Creating") { Thread sleep 10000 }
```

Comment:
Why not `await condition (getClusterStatus(clusterName) == "Running")`?

This method of waiting is undesirable for 2 reasons:
1. This method would succeed even if the cluster errors
2. There is no timeout - so if the UI status wasn't updated for some reason, this test would hang forever.